### PR TITLE
[Jetcaster] Migrate from M2 to M3.

### PR DESCRIPTION
**Pull request by arriolac**
Upgrading Jetcaster from M2 to M3. This migration changes one of the original purposes of the sample which was to demonstrate dynamic theming. We would still like to demonstrate dynamic theming but using a more modern approach using Material's [material-color-utilities](https://github.com/material-foundation/material-color-utilities).
Screenshot_20240319_131815

### CODE REVIEW

File: Jetcaster/app/src/main/java/com/example/jetcaster/ui/home/Home.kt [hunk 1]
```diff
@@ -403,28 +368,33 @@ fun HomeCategoryTabIndicator(
...
-        FollowedPodcastCarouselItem(
-            podcastImageUrl = podcast.imageUrl,
-            podcastTitle = podcast.title,
-            onUnfollowedClick = { onPodcastUnfollowed(podcast.uri) },
-            lastEpisodeDateText = lastEpisodeDate?.let { lastUpdated(it) },
-            modifier = Modifier
-                .padding(4.dp)
-                .fillMaxSize()
+    val lastIndex = items.size - 1
+    LazyRow(
```
Comment (bentrengrove) [inline Jetcaster/app/src/main/java/com/example/jetcaster/ui/home/Home.kt:378]: Why was this changed? I assume the old one was to achieve snapping behavior, which should be possible without HorizontalPager if you need. But also there is a new Carousel in M3 which might be good also, I haven't actually looked into it though

File: Jetcaster/app/src/main/java/com/example/jetcaster/ui/theme/Color.kt [hunk 2]
```diff
@@ -16,37 +16,31 @@
...
-    primary = Yellow800,
-    onPrimary = Color.Black,
-    primaryVariant = Yellow800,
-    secondary = Yellow800,
-    onSecondary = Color.Black,
-    error = Red300,
-    onError = Color.Black
-)
+// /**
+// * Return the fully opaque color that results from compositing [onSurface] atop [surface] with the
```
Comment (bentrengrove) [inline Jetcaster/app/src/main/java/com/example/jetcaster/ui/theme/Color.kt:27]: Delete?

**Review by bentrengrove — APPROVED** [review]

**Comment by bentrengrove** [comment]
The status bar color doesn't look correct in your screenshot, should be white.

File: Jetcaster/app/src/main/java/com/example/jetcaster/ui/theme/Color.kt (hunk 2 above)
Comment (arriolac) [reply Jetcaster/app/src/main/java/com/example/jetcaster/ui/theme/Color.kt:27]: Deleted. Thanks

**Comment by arriolac** [comment]
> The status bar color doesn't look correct in your screenshot, should be white.

That change is intentional — though the status bar should match `surface` color. Updating that

**Comment by bentrengrove** [comment]
Does the contrast of the text on the status bar color look correct now? Hard to tell from the code

**Comment by arriolac** [comment]
> Does the contrast of the text on the status bar color look correct now? Hard to tell from the code

Ahh good catch. I didn't notice the status bar text contrast. Looks like there was a bug in `JetcasterTheme`.
